src/main/java/edu/kis/vh/nursery/dataStructures/IntArrayStack.java

```diff
@@ -1,36 +1,40 @@
-package edu.kis.vh.nursery.list;
+package edu.kis.vh.nursery.dataStructures;
 
-public class IntArrayStack {
+public class IntArrayStack implements DataStructure {
 
-    private static final int EMPTY_INDEX = -1;
     private static final int STACK_MAX = 12;
     private final int[] numbers = new int[STACK_MAX];
 
     private int total = EMPTY_INDEX;
 
+    @Override
     public void countIn(int in) {
         if (!isFull())
-            numbers[++total] = in;
+            numbers[total++] = in;
     }
 
+    @Override
     public boolean callCheck() {
         return total == EMPTY_INDEX;
     }
 
+    @Override
     public boolean isFull() {
-        return total == (STACK_MAX - 1);
+        return total == STACK_MAX;
     }
 
+    @Override
     public int peekaboo() {
         if (callCheck())
             return EMPTY_INDEX;
         return numbers[total];
     }
 
+    @Override
     public int countOut() {
         if (callCheck())
             return EMPTY_INDEX;
-        return numbers[total--];
+        return numbers[--total];
```

Comment:
Changing EMPTY_INDEX value requires such fixes? It does not make sense. Incorrect fix.